### PR TITLE
Refactor get_users API to show proper swagger documentation

### PR DESCRIPTION
**Pull request by iamsdas**
Refactor the `get_users` API (under facility) to make the user role (`user_type`) filter visible in the swagger UI.

Fixes: #711

### CODE REVIEW

File: care/facility/api/viewsets/facility_users.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    serializer_class = UserAssignedSerializer
+    filterset_class = UserFilter
+    queryset = User.objects.all()
+    permission_classes = [IsAuthenticated]
+    filter_backends = [filters.DjangoFilterBackend]
+    ordering_fields = ["-last_login"]
+
+    def get_queryset(self):
+        try:
+            facility = Facility.objects.get(external_id=self.kwargs.get("facility_external_id"))
```
Comment (vigneshhari) [inline care/facility/api/viewsets/facility_users.py:32]: @gigincg does this endpoint need authorisation?

File: care/facility/api/viewsets/facility_users.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    user_type = filters.TypedChoiceFilter(choices=[(key, key) for key in User.TYPE_VALUE_MAP.keys()],
+                                          coerce=lambda role: User.TYPE_VALUE_MAP[role])
+
+    class Meta:
+        model = User
+        fields = []
+
+
+class FacilityUserViewSet(GenericViewSet, mixins.ListModelMixin):
+    lookup_field = "external_id"
```
Comment (vigneshhari) [inline care/facility/api/viewsets/facility_users.py:22]: leave this if you are not using the retrieve mixin

File: care/facility/api/viewsets/facility_users.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    serializer_class = UserAssignedSerializer
+    filterset_class = UserFilter
+    queryset = User.objects.all()
+    permission_classes = [IsAuthenticated]
+    filter_backends = [filters.DjangoFilterBackend]
+    ordering_fields = ["-last_login"]
```
Comment (vigneshhari) [inline care/facility/api/viewsets/facility_users.py:28]: do this by default

**Review by vigneshhari — CHANGES_REQUESTED** [review]

**Comment by vigneshhari** [comment]
you can also add swagger docs by using the fake schema generator that YASG provides

File: care/facility/api/viewsets/facility_users.py (hunk 2 above)
Comment (iamsdas) [reply care/facility/api/viewsets/facility_users.py]: updated it

**Review by vigneshhari — APPROVED** [review]

**Review by vigneshhari — APPROVED** [review]

**Pull request merged**